lib/handler/headers_util.c

```diff
@@ -4,26 +4,120 @@ static h2o_header_t *find_header(h2o_headers_t *headers, h2o_headers_command_t *
 {
     ssize_t index;
 
-    if (h2o_iovec_is_token(cmd->name)) {
-        index = h2o_find_header(headers, (void *)cmd->name, -1);
+    assert(cmd->num_args == 1);
+    if (h2o_iovec_is_token(cmd->args[0].name)) {
+        index = h2o_find_header(headers, (void *)cmd->args[0].name, -1);
     } else {
-        index = h2o_find_header_by_str(headers, cmd->name->base, cmd->name->len, -1);
+        index = h2o_find_header_by_str(headers, cmd->args[0].name->base, cmd->args[0].name->len, -1);
     }
     if (index == -1)
         return NULL;
     return headers->entries + index;
 }
 
+static int is_in_list(const char *base, size_t len, h2o_headers_command_t *cmd, int header)
+{
+    size_t i;
+    h2o_iovec_t name = h2o_iovec_init(base, len);
+    for (i = 0; i != cmd->num_args; ++i) {
+        if (header && h2o_iovec_is_token(cmd->args[i].name)) {
+            if (cmd->args[i].name->base == name.base) {
+                return 1;
+            }
+        } else {
+            if (h2o_memis(cmd->args[i].name->base, cmd->args[i].name->len, name.base, name.len))
+                return 1;
+        }
+    }
+    return 0;
+}
+
+enum {
+    UNSET,
+    UNSETUNLESS,
+};
+static void filter_cookie(h2o_mem_pool_t *pool, char **base, size_t *len, h2o_headers_command_t *cmd, int action)
```

Comment:
I think action can be determined by consulting `cmd->cmd`, and therefore that we do not need the action enum, and that `h2o_rewrite_headers` can directly call `filter_cookie` instead of using action-specific wrappers.